accepted/00000-minion-selective-action-blocking.md

```diff
@@ -0,0 +1,93 @@
+- Feature Name: minion_selective_action_blocking
+- Start Date: 2020-01-21
+- RFC PR: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+The goal is to selectively block certain action types from being executed on a Salt minion.
+
+# Motivation
+[motivation]: #motivation
+
+Main use cases:
+* Prevent certain action types when a product is installed on a minion, e.g. when a minion is a CaaSP node some actions shouldn't be allowed.
+* Feature parity with traditional clients. System locking is available for traditional clients but no such functionality exists for Salt minions.
+
+Usability:
+* Salt has a blackout mechanism. This is used for the first version of CaaSP support but it's not flexible enough for Uyuni's use case. Uyuni makes heavy use of `state.apply` to execute actions. Salt blackout allows whitelisting which modules are allowed during blackout. In the case of Uyuni, whitelisting `state.apply` means allowing basically all actions. Additionally the user experience of using Salt blackout is not great, the user can schedule any actions from the UI/API but then gets an error message at execution time that the action is not allowed by Salt.
+
+
+# Detailed design
+[design]: #detailed-design
+
+Traditional clients can be locked by the user. When locked, only certain action types are allowed. Actions that change the state of a system are blocked (package install, upgrade, apply highstate, etc) while read-only actions are allowed (hardware refresh, package profile refresh, etc).
```

Comment:
There is one notable example that works for traditional systems even when locked: remote commands.
Do you think we should enable it for minion system locked as well? In sake of feature parity, I would say we should mimic the same behavior.

Reply:
Yes, I think for now we should be consistent with the traditional clients.

Reply:
Please add this bit of information as it is a notable exception IMHO